engine/execution/ingestion/engine.go

```diff
@@ -641,6 +645,10 @@ func (e *Engine) executeBlock(ctx context.Context, executableBlock *entity.Execu
 	if err != nil {
 		e.log.Err(err).Msg("failed in process block's children")
 	}
+
+	if e.executionDataPruner != nil {
+		e.executionDataPruner.NotifyFulfilledHeight(executableBlock.Height())
```

Comment:
> Since we persist execution data asynchronously

As mentioned above, execution data is actually persisted synchronously, in that the Provider doesn't return until it has been persisted or an error is encountered.